dev.gbstudio.gb-studio.yaml

```diff
@@ -0,0 +1,44 @@
+id: dev.gbstudio.gb-studio
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '22.08'
+command: gb-studio
+finish-args:
+  - --socket=fallback-x11
+  - --filesystem=host:rw
+  - --socket=wayland
+  - --device=dri
+  - --share=ipc
+
+modules:
+  - name: gb-studio
+    buildsystem: simple
+    build-commands:
+      - chmod +x gb-studio-linux.AppImage
+      - ./gb-studio-linux.AppImage --appimage-extract
+      - cp -r squashfs-root/usr/lib/gb-studio $FLATPAK_DEST/lib
+      - install -Dm755 gb-studio -t $FLATPAK_DEST/bin
+      - install -Dm644 dev.gbstudio.gb-studio.png -t $FLATPAK_DEST/share/icons/hicolor/128x128/apps
+      - install -Dm644 dev.gbstudio.gb-studio.desktop -t $FLATPAK_DEST/share/applications
+      - install -Dm644 dev.gbstudio.gb-studio.metainfo.xml -t $FLATPAK_DEST/share/metainfo
+    sources:
+      - type: file
+        url: https://github.com/chrismaltby/gb-studio/releases/download/v3.1.0/gb-studio-linux.AppImage
```

Comment:
Wouldn't it be possible to build from source rather than unpack the AppImage?

Reply:
It is a electron App. Building from source without Internet is very hard, so almost all Electron Apps (especially the more complex) just uses a prebuild binary.

Reply:
I think this would be the tool to use to generate the dependencies file:
https://github.com/flatpak/flatpak-builder-tools/tree/master/node

And this is how it's used in one of my own packages:
https://github.com/flathub/io.github.neil_morrison44.pocket-sync/blob/master/io.github.neil_morrison44.pocket-sync.yaml#L36-L37

Reply:
I tried it an got this error:
```
error Couldn't find any versions for "@reforged/maker-appimage" that matches "^1.1.3" in our cache (possible versions are ""). This is usually caused by a missing entry in the lockfile, running Yarn without the --offline flag may help fix this issue.
```

If you search for `maker-appimage `in the [lockfile](https://github.com/chrismaltby/gb-studio/blob/v3.1.0/yarn.lock), you will find nothing.

Reply:
> I tried it an got this error:
> 
> ```
> error Couldn't find any versions for "@reforged/maker-appimage" that matches "^1.1.3" in our cache (possible versions are ""). This is usually caused by a missing entry in the lockfile, running Yarn without the --offline flag may help fix this issue.
> ```

```
$ flatpak-node-generator yarn yarn.lock
```
on the 3.1.0 tag generated this file: 
[generated-sources.json.txt](https://github.com/flathub/flathub/files/11089531/generated-sources.json.txt)

> If you search for `maker-appimage `in the [lockfile](https://github.com/chrismaltby/gb-studio/blob/v3.1.0/yarn.lock), you will find nothing.

It's mentioned on the `develop` branch:
https://github.com/chrismaltby/gb-studio/blob/develop/yarn.lock#L1549

Did you run the script on the correct branch?

Reply:
I ran it on `3.1.0`, as this is the version I want to package. I have now tried it with `develop`. It now has it in the `generated-socures.json`, but I still get the same error, so I guess my manifest is wrong.

```yaml
id: dev.gbstudio.gb-studio
runtime: org.freedesktop.Platform
runtime-version: '22.08'
sdk: org.freedesktop.Sdk
base: org.electronjs.Electron2.BaseApp
base-version: '22.08'
#command: gb-studio
sdk-extensions:
  - org.freedesktop.Sdk.Extension.node18
finish-args:
  - --socket=fallback-x11
  - --filesystem=host:rw
  - --socket=wayland
  - --device=dri
  - --share=ipc

modules:
  - name: gb-studio
    buildsystem: simple
    build-options:
        append-path: /usr/lib/sdk/node18/bin
        env:
            CHROMEDRIVER_SKIP_DOWNLOAD: 'true'
            npm_config_nodedir: '/usr/lib/sdk/node18'
            npm_config_offline: 'true'
            XDG_CACHE_HOME: /run/build/gb-studio/flatpak-node/cache
    build-commands:
      - HOME=$PWD yarn config --offline set yarn-offline-mirror $FLATPAK_BUILDER_BUILDDIR/flatpak-node/yarn-mirror
      - yarn --offline
      - yarn run --offline make
      - install -Dm644 dev.gbstudio.gb-studio.png -t $FLATPAK_DEST/share/icons/hicolor/512x512/apps
      - install -Dm644 dev.gbstudio.gb-studio.desktop -t $FLATPAK_DEST/share/applications
      - install -Dm644 dev.gbstudio.gb-studio.metainfo.xml -t $FLATPAK_DEST/share/metainfo
    sources:
      - type: archive
        archive-type: tar-gzip
        url: https://api.github.com/repos/chrismaltby/gb-studio/tarball/v3.1.0
        sha256: a2ceefd4fb49e0d0f1f01268cf08feb761f7d9564d302ff08bb06749cca613d1
        x-checker-data:
            type: json
            url: https://api.github.com/repos/chrismaltby/gb-studio/releases/latest
            version-query: .tag_name
            url-query: .tarball_url
      - generated-sources.json
      - type: script
        dest-filename: gb-studio
        commands:
          - zypak-wrapper /app/lib/gb-studio/gb-studio
      - type: file
        path: dev.gbstudio.gb-studio.png
      - type: file
        path: dev.gbstudio.gb-studio.desktop
      - type: file
        path: dev.gbstudio.gb-studio.metainfo.xml
```

Reply:
> Wouldn't it be possible to build from source rather than unpack the AppImage?

FYI https://github.com/flathub/dev.gbstudio.gb-studio/pull/1